### PR TITLE
pfc_unit_and_unpause

### PR DESCRIPTION
**Pull request by Rangababu-R**
pfc testcases unit test and upause scripts

### CODE REVIEW

File: tests/pfc/test_pfc_unpause.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                            dst=flow.Pattern('00:AB:BC:AB:BC:AB'),
+                            pfc_queue=flow.Pattern(str(prio))
+                        )
+                    ),
+                    flow.Header(
+                        flow.Ipv4(
+                            src=flow.Pattern('1.1.1.2'),
+                            dst=flow.Pattern('1.1.1.1'),
+                            priority=flow_ipv4.Priority(
+                                flow_ipv4.Dscp(flow.Pattern(str(prio)))
```
Comment (ashutshkumr) [inline tests/pfc/test_pfc_unpause.py:75]: Please provide correct PHB_CS* value here.

File: tests/pfc/test_pfc_unpause.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                        class_enable_vector=flow.Pattern('FF'),
+                        control_op_code=flow.Pattern('0101'),
+                        pause_class_3=flow.Pattern('FFFF'),
+                        pause_class_4=flow.Pattern('FFFF')
+                    )
+                )
+            ],
+            duration=flow.Duration(flow.FixedSeconds(seconds=10)),
+            size=flow.Size(size),
+            rate=flow.Rate(value=packets / 10, unit='pps')
```
Comment (ashutshkumr) [inline tests/pfc/test_pfc_unpause.py:109]: let's use 50% of line rate here, other 50% will be shared by unpause flow when its time comes.

File: tests/pfc/test_pfc_unpause.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                        pause_class_3=flow.Pattern('0000'),
+                        pause_class_4=flow.Pattern('0000')
+                    )
+                )
+            ],
+            duration=flow.Duration(flow.FixedSeconds(
+                seconds=10, delay=(10**9) * 10, delay_unit='nanoseconds'
+            )),
+            size=flow.Size(size),
+            rate=flow.Rate(value=packets / 10, unit='pps')
```
Comment (ashutshkumr) [inline tests/pfc/test_pfc_unpause.py:134]: should be 50% of line rate

File: tests/pfc/test_pfc_unpause.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    un_pause = [
+        f['frames_rx'] for f in flow_results if f['name'] == 'rx_pfc_unpause'
+    ][0]
+    ok = []
+
+    for fl in flow_results:
+        if fl['name'] == 'rx_unpause' or fl['name'] == 'rx_pause':
+            continue
+        if pause < packets and pause > 0 and check_for_pause and \
+           "tx_prio_" in fl['name']:
```
Comment (ashutshkumr) [inline tests/pfc/test_pfc_unpause.py:175]: following is more pedantic.
```python
fl['name'].startswith('tx_prio')
```

File: tests/pfc/test_pfc_unpause.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+    priorities = [
+        'pfc_class_{}_frames_rx'.format(p) for p in lossless_priorities
+    ]
+    for stat in p_stats['raw_tx']:
+        if (check_for_pause is False) and (stat in priorities):
+            ok.append(
+                p_stats['raw_tx'][stat] == packets * 2
+            )
+
+    return all(ok)
```
Comment (ashutshkumr) [inline tests/pfc/test_pfc_unpause.py:196]: Need to fix logical issue with `ok`. Need to ensure `ok` contains only given number of flags. i.e. I don't want `all(ok)` to return True when it's equal to `[True]`.

File: ixnetwork_open_traffic_generator/ixnetworkapi.py [hunk 6]
```diff
@@ -261,7 +261,7 @@ def _connect(self):
         """Connect to an IxNetwork API Server.
         """
         if self._assistant is None:
-            platform = TestPlatform(self._address, rest_port=self._password)
+            platform = TestPlatform(self._address, rest_port=self._port)
```
Comment (ashutshkumr) [inline ixnetwork_open_traffic_generator/ixnetworkapi.py:264]: Is it relevant to this PR ? if not, please include this as part of separate PR (linked to an issue).

**Review by ashutshkumr — CHANGES_REQUESTED** [review]
Please check the review comments. We also need to include following:
- assert PFC stats in PFC pause and unpause test scripts
- @ANISH-GOTTAPU maybe you need to take care of this - we need to validate L1 conig for global and PFC pause.

File: tests/pfc/test_pfc_unpause.py (hunk 2 above)
Comment (Rangababu-R) [reply tests/pfc/test_pfc_unpause.py]: changed to 50%

File: tests/pfc/test_pfc_unpause.py (hunk 4 above)
Comment (Rangababu-R) [reply tests/pfc/test_pfc_unpause.py]: modified

File: tests/pfc/test_pfc_unpause.py (hunk 5 above)
Comment (Rangababu-R) [reply tests/pfc/test_pfc_unpause.py]: made to fixed number of flags

File: ixnetwork_open_traffic_generator/ixnetworkapi.py (hunk 6 above)
Comment (Rangababu-R) [reply ixnetwork_open_traffic_generator/ixnetworkapi.py:264]: raise one more PR for it
https://github.com/open-traffic-generator/ixnetwork/pull/185/commits/a1ed8dc03b51b1e40bd80ac3738fbb4c0efc5a4f

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**